//  Copyright Authors DustinKsi@ Youibot 2019. All Rights Reserved.
// 
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
// 
//  http://www.apache.org/licenses/LICENSE-2.0
// 
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.

syntax = "proto3";

package DsFMapping.Extraction;

enum ExtractionBrand {
    Intensity = 0;
    Corner = 1;
}

message ExtractionOptions {
    ExtractionBrand brand = 1;

    bool use_adaptive_intensity_threshold = 2;
    int32 intensity_down_threshold = 3;
    int32 intensity_median_filter_param = 4;

    string corner_search_method = 5; 
    double corner_curve_thresold = 6;
}
